clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -730,6 +732,41 @@ public void testFlushCompleteSendOfInflightBatches() {
         }
     }
 
+    private static Double getMetricValue(final KafkaProducer<?, ?> producer, final String name) {
+        Metrics metrics = producer.metrics;
+        Metric metric =  metrics.metric(metrics.metricName(name, "producer-metrics"));
+        return (Double) metric.metricValue();
+    }
+
+    @Test
+    public void testFlushMeasureLatency() {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9000");
+
+        Time time = new MockTime(1);
+        MetadataResponse initialUpdateResponse = RequestTestUtils.metadataUpdateWith(1, singletonMap("topic", 1));
+        ProducerMetadata metadata = newMetadata(0, Long.MAX_VALUE);
+
+        MockClient client = new MockClient(time, metadata);
+        client.updateMetadata(initialUpdateResponse);
+
+        try (KafkaProducer<String, String> producer = kafkaProducer(
+            configs,
+            new StringSerializer(),
+            new StringSerializer(),
+            metadata,
+            client,
+            null,
+            time
+        )) {
+            producer.flush();
+            double first = getMetricValue(producer, "flush-time-total");
+            assertTrue(first > 999999.0);
```

Comment:
It's using mock time, so the value here is well-known (should be 1 second). I'm using > rather than equalTo because I don't want the test to fail spuriously on floating point rounding errors. It would probably be better to use [isCloseTo](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/number/IsCloseTo.html)

Reply:
Ah actually this doesn't work because the mock time is passed to and used from the other client threads - so the value is not predictable. So the best we can do is assert that at least one tick (1000000 nanoseconds has passed). I'll update the test.